.github/workflows/test.sh

```diff
@@ -89,9 +89,42 @@ if [[ $PLATFORM == linux || $PLATFORM == darwin ]]; then
     fi
 fi
 
+## `spin test` test cases
 ptest test command runs
 prun spin test
 
+ptest does gcov test generate coverage files
+prun spin test --gcov
+matching_files=$(find . -type f -name "*.gcda")
+if [ -z "$matching_files" ]; then
+    perror Coverage files did not get generated
+    exit 1
+else
+    echo "Yes"
+fi
+
+# TODO: Test other coverage reports
```

Comment:
TODO: I'll cover this as a part of https://github.com/scientific-python/spin/issues/154 to avoid bloat here